scalafix-cli/src/main/scala/scalafix/cli/CliRunner.scala

```diff
@@ -415,16 +416,16 @@ object CliRunner {
           .notOk
     }
 
-    val diffDisable: Option[DiffDisable] = {
+    val diffDisable: Configured[Option[DiffDisable]] = {
       if (cli.diff || cli.diffBranch.nonEmpty) {
         val baseBranch = cli.diffBranch.getOrElse("master")
-        Some(DiffDisable(common.workingPath.toNIO, baseBranch))
+        DiffDisable(common.workingPath.toNIO, baseBranch).map(Some(_))
       } else {
-        None
+        Configured.Ok(None)
       }
     }
 
-    val fixFiles: Configured[Seq[FixFile]] =
+    val fixFiles: Configured[Seq[FixFile]] = diffDisable.andThen(diffDisable0 =>
       resolvedPathMatcher.andThen { pathMatcher =>
```

Comment:
You can use `diffDisable.product(resolvedPathMatcher)` or `|@|` instead of product